cache/compression.go

```diff
@@ -94,20 +94,41 @@ func (s *SnappyCache) Add(ctx context.Context, key string, value []byte, ttl tim
 
 // GetMulti implements Cache.
 func (s *SnappyCache) GetMulti(ctx context.Context, keys []string, opts ...Option) map[string][]byte {
-	found := s.next.GetMulti(ctx, keys, opts...)
+	result, err := s.GetMultiWithError(ctx, keys, opts...)
+
+	if err != nil {
+		joinedErrs, ok := err.(interface{ Unwrap() []error })
```

Comment:
I don't think anything this complicated is needed, just logging any error at `warn` is fine. I don't expect that this is common at all.